### PR TITLE
feat: Helm - custom service account creation and management

### PR DESCRIPTION
**Pull request by wiktor2200**
### SUMMARY

Possibility to create custom service account in helm chart.

### TESTING INSTRUCTIONS

I've tested all possible changes on parameters:
* serviceAccountName
* serviceAccount.create
All conditionals in `_helpers` file works as expected.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/17879
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by amitmiran137** [comment]
Could you bump version of the chart manually ?

**Comment by wiktor2200** [comment]
Sure! It's done :)

**Comment by wiktor2200** [comment]
@amitmiran137 there is something wrong with workflow. It shows error in code which I didn't change.
BTW, who can be assigned as reviewer for this PR? Are you maintainer of helm charts?

File: helm/superset/values.yaml
Comment (przemyslawandruszewski) [inline helm/superset/values.yaml:31]: How about:
1. Remove "serviceAccountName"
2. creating serviceAccount sestion as you did with two properties:
- create as you did
- name

According to: https://helm.sh/docs/chart_best_practices/rbac/#yaml-configuration

**Review by przemyslawandruszewski — COMMENTED** [review]
Would you please also put the serviceAccountName to "init-job" also?

File: helm/superset/values.yaml
Comment (wiktor2200) [reply helm/superset/values.yaml:31]: I've thought about it. That was even my first choice, but then I've noticed that, there was `ServiceAccountName` variable added in the past #15340, so I've implemented new feature as enhancement beside the current solution.
I don't want to destroy any configs that can exist somewhere.
@mvoitko and @craig-rueda can you give some more feedback about this?

**Comment by wiktor2200** [comment]
@przemyslawandruszewski I've added support for custom service account in init-job in this commit: https://github.com/apache/superset/pull/17880/commits/372fed158346767d24305dd185bed448a295107d

File: helm/superset/values.yaml
Comment (przemyslawandruszewski) [reply helm/superset/values.yaml:31]: Yeps you are right, we would break backward compatibility but on the other site, we could maintain both for sometime and prepare a communication about that right now.

**Comment by amitmiran137** [comment]
> @amitmiran137 there is something wrong with workflow. It shows error in code which I didn't change.
> BTW, who can be assigned as reviewer for this PR? Are you maintainer of helm charts?

Yes we became aware lately 
It seems to be broken also in master 
We are trying to solve this

File: helm/superset/values.yaml
Comment (wiktor2200) [reply helm/superset/values.yaml:31]: If someone have blocked version on helm in their definition it's OK, but when someone is using "latest" it will brake down deployment (or dependent components).
Supporting two solutions for a while should be quite easy (I'll just add one more conditions in `_helpers`) but this could cause some caveats such as overwriting values if `serviceAccountName` and `serviceAccount.name` would be defined in the same time. :thinking:

**Comment by przemyslawandruszewski** [comment]
@wiktor2200 I cannot find any template for creating service account when serviceAccount.create == true. Could you validate whether you commited all the stuff?

**Comment by wiktor2200** [comment]
@przemyslawandruszewski you got right, thanks for notifying that. 
IDK how this happened, but I've not commited this file. Maybe I run git add with some regexp and it has not been added :man_shrugging:

File: helm/superset/templates/service-account.yaml
Comment (przemyslawandruszewski) [inline helm/superset/templates/service-account.yaml:5]: Wouldn't that be enough to use:
{{ include "superset.serviceAccountName" . }}

**Comment by przemyslawandruszewski** [comment]
The rest is fine for me, let's leave final decision for @craig-rueda :)

File: helm/superset/templates/service-account.yaml
Comment (wiktor2200) [reply helm/superset/templates/service-account.yaml]: That's good idea, I've written whole conditional before working on `_helpers` :)
It's done!

File: helm/superset/values.yaml
Comment (craig-rueda) [reply helm/superset/values.yaml:31]: I think it LGTM. If people are using "latest" in any env that matters, that's generally a bad practice. Using the default of `create: false` seems reasonable to me.

**Review by craig-rueda — APPROVED** [review]

**Comment by wiktor2200** [comment]
@craig-rueda Pipeline failed because of missing license header. I've added it, could you re-approve? :)

**Review by przemyslawandruszewski — APPROVED** [review]

**Comment by przemyslawandruszewski** [comment]
@wiktor2200 looks like you can merge :)

**Comment by wiktor2200** [comment]
Unfortunately I can't. :( It has to be clicked by someone with write access to repo.
screen

**Comment by przemyslawandruszewski** [comment]
> Unfortunately I can't. :( It has to be clicked by someone with write access to repo. screen

Sorry then :) So the request goes to @craig-rueda :)

**Pull request merged**
